[ENG-7283] fixed publication doi related error

ihorsokhanexoft opened the pull request:
## Purpose

An unexpected error was raised because incorrect Publication DOI was passed

## Changes

If possible try to strip redundant data and save the correct format

## Ticket

https://openscience.atlassian.net/browse/ENG-7283

brianjgeiger commented on api/preprints/serializers.py:
```diff
@@ -493,6 +496,16 @@ def require_admin_permission():
+                    validate_doi(article_doi)
+                except ValidationValueError:
+                    # try to strip redundant data and save it in the correct format
+                    matches = re.findall('10\\.\\S*\\/', article_doi)
```
Would it make more sense to strip the non-doi stuff out before trying to validate the doi, then running it through `validate_doi` afterward? That way you don't have to validate it twice, it just validates through the validate function.

ihorsokhanexoft replied:
Yep, I tried this approach but some tests fail because this line
https://github.com/CenterForOpenScience/osf.io/blob/1d5c7e4954084666321413c5314cac87e4a94675/api/preprints/serializers.py#L493
Will fail as `stripped article doi != doi` as `doi = 10.1234/test/FK2osf.io/{preprint._id}/ `for example, but `article_doi = 10.1234/test`. Even if I strip doi from that method, it'll be `10.1234/test/FK2osf.io/` that isn't equal to entered `10.1234/test` article doi

So, I would have to change if statement to `doi.startswith(article_doi)` but wasn't sure it's always correct
Will try to modify that statement

ihorsokhanexoft replied:
Okay, I see. We also need another regex for stripping the correct doi string. Working on it

brianjgeiger reviewed (changes requested):
One suggested change to logic.

ihorsokhanexoft commented:
I created a new regex to fetch the whole correct part of doi string. I see that in the app we show format "10.xxxx/xxxxx" but in the validator "it should start with 10 and have at least 1 slash". So I accept both numbers and letters and any number of them, including slashes

ihorsokhanexoft commented:
And also I noticed preprint provider wasn't set thus some tests had 'None/osf.io/{guid}/ doi string. Added it and replaced 1234 by 1235 to avoid associating error
